compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -346,7 +346,7 @@ object Parsers {
             false // it's a statement that might be legal in an outer context
           else
             in.nextToken() // needed to ensure progress; otherwise we might cycle forever
-            skip()
+            skip(stopAtComma=false)
```

Comment:
Here, we are explicitly looking to skip to a place that should end a statement, which a `,` never will.